pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/SinkConfigUtils.java

```diff
@@ -416,7 +416,7 @@ public static SinkConfig validateUpdate(SinkConfig existingConfig, SinkConfig ne
                 if (!existingConfig.getInputSpecs().containsKey(topicName)) {
                     throw new IllegalArgumentException("Input Topics cannot be altered");
                 }
-                if (!consumerConfig.equals(existingConfig.getInputSpecs().get(topicName))) {
+                if (consumerConfig.isRegexPattern() != existingConfig.getInputSpecs().get(topicName).isRegexPattern()) {
                     throw new IllegalArgumentException("Input Specs mismatch");
```

Comment:
Can we modify the message of the exception to be more specific?  That would help a lot in debugging

Reply:
Changed